src/pini.cpp

```diff
@@ -0,0 +1,32 @@
+#include "pini.hpp"
+#include "util.hpp"
+#include <cstddef>
+#include <sstream>
+#include <vector>
+
+namespace pn {
+
+std::unordered_map<std::string, std::string> pn::reader::get_pairs() {
+  return reader::key_value_pairs;
+}
+bool pn::reader::read_file(std::filesystem::path const &filename) {
+
+  file_lines = util::get_lines(filename);
+  if (file_lines.empty()) {
+    return false;
+  }
+  key_value_pairs = util::insert_pairs(file_lines);
+
+  return true;
```

Comment:
I'm not really seeing why the return type is `bool` when the function always returns `true` at the end of the function call - this might as well be `void` IMHO